apps/comments/views.py

```diff
@@ -44,3 +51,24 @@ def form_valid(self, form):
         response = super().form_valid(form)
         messages.success(self.request, 'Thank you the comment was reported.')
         return response
+
+
+class CommentUpdateView(LoginRequiredMixin, UpdateView):
+    model = Comment
+    form_class = CommentForm
+    slug_url_kwarg = 'id'
+    slug_field = 'id'
+    template_name = 'comments/comment_update.html'
+
+    def get_success_url(self):
+        return self.object.resource.get_absolute_url()
+
+    def get(self, request, *args, **kwargs):
+        response = super().get(request, *args, **kwargs)
+        if self.object.created_by != self.request.user:
```

Comment:
Only problem here is that it doesn't protect POST requests.

It might be easier to tweak the queryset (has to be the get_queryset method though) to limit for comments created_by self.request.user? That way it'll apply to both GET and POST requests.

Reply:
Changed the `def get_object` method